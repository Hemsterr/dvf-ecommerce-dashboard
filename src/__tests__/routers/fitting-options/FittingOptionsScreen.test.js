// Libs
import React from 'react'
import { render, fireEvent, wait } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'

// Component
import FittingOptions from '../../../routers/fitting-options'
import AppContext from '../../../contexts'

const store = {
  dispatch: jest.fn(),
  state: {
    fittingOption: '',
  },
}

const FittingOptionsScreen = store => (
  <AppContext.Provider value={store}>
    <Router>
      <FittingOptions />
    </Router>
  </AppContext.Provider>
)

describe('FittingOptions screen', () => {
  const { container, rerender } = render(FittingOptionsScreen(store))

  test('Render the FittingOptions screen with default props', async () => {
    await wait()
    expect(container).toMatchSnapshot()
  })

  describe('Checking events', () => {
    test('should be rendered when click button', async () => {
      rerender(FittingOptionsScreen(store))
      const bookingItem = container.querySelector('.bookingItem__fitting-kit')
      fireEvent.click(bookingItem)
      await wait()
      rerender(
        FittingOptionsScreen({
          ...store,
          state: {
            fittingOption: 'fitting-kit',
          },
        })
      )
    })
  })
})
