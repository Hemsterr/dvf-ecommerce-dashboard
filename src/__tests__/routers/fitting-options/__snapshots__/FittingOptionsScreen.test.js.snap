// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FittingOptions screen Render the FittingOptions screen with default props 1`] = `
<div>
  <div
    class="fitting-options"
  >
    <p
      class="fitting-options__title"
    >
      Please select an option below to have your garment altered:
    </p>
    <div
      class="fitting-options__wrapper"
    >
      <div
        class="bookingItem bookingItem__fitting-kit"
        role="presentation"
      >
        <div>
          <span
            class="bookingItem__icon"
          />
        </div>
        <p
          class="bookingItem__title"
        >
          I would like to fit myself using Hemster's Fitting Kit
        </p>
      </div>
      <div
        class="bookingItem bookingItem__fit-stylist"
        role="presentation"
      >
        <div>
          <span
            class="bookingItem__icon"
          />
        </div>
        <p
          class="bookingItem__title"
        >
          I would like to book an appointment to meet with a Fit Stylist virtually
        </p>
      </div>
      <div
        class="bookingItem bookingItem__in-home-stylist"
        role="presentation"
      >
        <div>
          <span
            class="bookingItem__icon"
          />
        </div>
        <p
          class="bookingItem__title"
        >
          I would like to book an In-Home Appointment with a Fit Stylist
        </p>
      </div>
    </div>
    <div
      class="fitting-options__need-repairs"
    >
      <p
        class="fitting-options__title fitting-options__need-repairs__title"
      >
        Need repairs?
      </p>
      <p
        class="fitting-options__description"
      >
        Please 
        <i>
          click here
        </i>
         for Request Form
      </p>
      <span
        class="fitting-options__description fitting-options__policy"
      >
        DVF may wish to replace your garment or provide complimentary servicing on your garment.
         
      </span>
      <span
        class="fitting-options__description fitting-options__policy"
      >
        Once we receive status of your request, we will send email confirmation with more information on how to proceed with your repairs!
      </span>
    </div>
  </div>
</div>
`;
